src/Microsoft.OpenApi.OData.Reader/Operation/SingletonGetOperationHandler.cs

```diff
@@ -86,45 +94,49 @@ protected override void SetParameters(OpenApiOperation operation)
 
-            if (Context.Settings.ShowLinks)
-            {
-                links = Context.CreateLinks(entityType: Singleton.EntityType, entityName: Singleton.Name,
-                        entityKind: Singleton.ContainerElementKind.ToString(), path: Path, parameters: PathParameters);
-            }
+                if (Context is {Settings.EnableDerivedTypesReferencesForResponses: true})
+                {
+                    schema = EdmModelHelper.GetDerivedTypesReferenceSchema(Singleton.EntityType, Context.Model, _document);
+                }
+
+                if (Context is {Settings.ShowLinks: true} && Path is not null)
+                {
+                    links = Context.CreateLinks(entityType: Singleton.EntityType, entityName: Singleton.Name,
+                            entityKind: Singleton.ContainerElementKind.ToString(), path: Path, parameters: PathParameters);
+                }
 
-            schema ??= new OpenApiSchemaReference(Singleton.EntityType.FullName(), _document);
+                schema ??= new OpenApiSchemaReference(Singleton.EntityType.FullName(), _document);
 
-            operation.Responses = new OpenApiResponses
-            {
+                operation.Responses = new OpenApiResponses
                 {
-                    Context.Settings.UseSuccessStatusCodeRange ? Constants.StatusCodeClass2XX : Constants.StatusCode200,
-                    new OpenApiResponse
                     {
-                        Description = "Retrieved entity",
-                        Content = new Dictionary<string, OpenApiMediaType>
+                        Context?.Settings.UseSuccessStatusCodeRange ?? false ? Constants.StatusCodeClass2XX : Constants.StatusCode200,
+                        new OpenApiResponse
                         {
+                            Description = "Retrieved entity",
+                            Content = new Dictionary<string, OpenApiMediaType>
                             {
-                                Constants.ApplicationJsonMediaType,
-                                new OpenApiMediaType
                                 {
-                                    Schema = schema
+                                    Constants.ApplicationJsonMediaType,
+                                    new OpenApiMediaType
+                                    {
+                                        Schema = schema
+                                    }
                                 }
-                            }
-                        },
-                        Links = links
+                            },
+                            Links = links
+                        }
                     }
-                }
-            };
+                };
+            }
 
-    		operation.AddErrorResponses(Context.Settings, _document, false);
+    		if (Context is not null)
+                operation.AddErrorResponses(Context.Settings, _document, false);
```

Comment:
Nit: Not sure whether this line is properly indented

Reply:
just double checked, it is. Resolving